src/plugins/layout/stateprovider.ts

```diff
@@ -94,30 +112,83 @@ export function createElementLayoutState(element: Element): IElementLayoutState
   if (rect) {
-    let styles = window.getComputedStyle(element);
-
-    elementState.layout = {
+    layout = {
       x: Math.round(rect.left),
       y: Math.round(rect.top),
       width: Math.round(rect.width),
       height: Math.round(rect.height)
     };
+  }
+  return layout;
+}
 
-    // Check if scroll is possible
-    if (styles["overflow-x"] === "auto"
-                              || styles["overflow-x"] === "scroll"
-                              || styles["overflow-x"] === "hidden"
-                              || styles["overflow-y"] === "auto"
-                              || styles["overflow-y"] === "scroll"
-                              || styles["overflow-y"] === "hidden") {
-      elementState.layout.scrollX = Math.round(element.scrollLeft);
-      elementState.layout.scrollY = Math.round(element.scrollTop);
+function getAttributes(element) {
+  let elementAttributes = element.attributes;
+  let tagName = element.tagName;
+  let stateAttributes: IAttributes = {};
+  for (let i = 0; i < elementAttributes.length; i++) {
+    let attr = elementAttributes[i];
+    let attrName = attr.name.toLowerCase();
+
+    // If it's an image and configuration disallows capturing images then skip src attribute
+    if (tagName === "IMG" && !config.showImages && attrName === "src") {
+      continue;
+    }
+
+    // If we are masking text, also mask it from input boxes as well as alt description
+    if (!config.showText && attributeMaskList.indexOf(attrName) >= 0) {
+      stateAttributes[attr.name] = attr.value.replace(/\S/gi, "*");
+    } else {
+      stateAttributes[attr.name] = attr.value;
     }
   }
+  return stateAttributes;
+}
 
-  return elementState;
+function getStyles(element) {
+    let computed = window.getComputedStyle(element);
+    let style = {};
+
+    if (defaultColor.length === 0) {
+      defaultColor = computed[Styles.Color];
+    }
+
+    // Send computed styles, if relevant, back to server
+    if (match(computed[Styles.Visibility], ["hidden", "collapse"])) {
+      style[Styles.Visibility] = computed[Styles.Visibility];
+    }
+
+    if (match(computed[Styles.OverflowX], ["auto", "scroll", "hidden"])) {
+      style[Styles.OverflowX] = computed[Styles.OverflowX];
+    }
+
+    if (match(computed[Styles.OverflowY], ["auto", "scroll", "hidden"])) {
+      style[Styles.OverflowY] = computed[Styles.OverflowY];
+    }
+
+    if (computed[Styles.BackgroundImage] !== "none") {
+      style[Styles.BackgroundImage] = computed[Styles.BackgroundImage];
+    }
+
+    if (!match(computed[Styles.BackgroundColor], ["rgba(0, 0, 0, 0)", "transparent"])) {
+      style[Styles.BackgroundColor] = computed[Styles.BackgroundColor];
+    }
+
+    if (computed[Styles.Color] !== defaultColor) {
+      style[Styles.Color] = computed[Styles.Color];
+    }
+
+    return Object.keys(style).length > 0 ? style : null;
+}
+
+function match(variable: string, values: string[]): boolean {
+  for (let i = 0; i < values.length; i++) {
```

Comment:
values.indexOf(variable) > -1

Reply:
Kept the function but replaced the body with your suggestion. I forgot we don't have to worry about older browsers here.